Objects/frameobject.c

```diff
@@ -1088,13 +1094,12 @@ _PyFrame_LocalsToFast(_PyInterpreterFrame *frame, int clear)
 void
 PyFrame_LocalsToFast(PyFrameObject *f, int clear)
 {
-    if (f == NULL || _PyFrame_GetState(f) == FRAME_CLEARED) {
-        return;
+    if (f && f->f_fast_as_locals && _PyFrame_GetState(f) != FRAME_CLEARED) {
```

Comment:
I would prefer to move these tests inside _PyFrame_LocalsToFast(), only keep the "f == NULL" here, for the public C API. And also set f_fast_as_locals in _PyFrame_LocalsToFast().

Otherwise, calling _PyFrame_LocalsToFast() makes f_fast_as_locals inconsistent, no?

Reply:
We can't do that as `_PyFrame_LocalsToFast()` takes a `_PyInterpreterFrame` not a frame object.

Only the interpreter calls `_PyFrame_LocalsToFast()`, to implement `import *`, and always calls `_PyFrame_FastToLocals()` in a non-destructive way.

Reply:
Oh ok, I see.

Reply:
Good point about `f_fast_as_locals`.
It won't be inconsistent, but if it is modified outside of a tracing function, we must reset those changes before tracing.
I've fixed that.